libs/labelbox/src/labelbox/schema/dataset.py

```diff
@@ -912,3 +913,80 @@ def _convert_items_to_upsert_format(self, _items):
             }  # remove None values
             _upsert_items.append(DataRowUpsertItem(payload=item, id=key))
         return _upsert_items
+    
+    def set_iam_integration(self, iam_integration: Union[str, IAMIntegration, None] = None) -> Optional[IAMIntegration]:
+          """
+          Sets the IAM integration for the dataset. IAM integration is used to sign URLs for data row assets.
+
+          >>>    # Get all IAM integrations
+          >>>    iam_integrations = client.get_organization().get_iam_integrations()
+          >>>    
+          >>>    # Get IAM integration id
+          >>>    iam_integration_id = [integration.uid for integration
+          >>>      in iam_integrations
+          >>>      if integration.name == "My S3 integration"][0]
+          >>>
+          >>>   # Set IAM integration for integration id
+          >>>   dataset.set_iam_integration(iam_integration_id)
+          >>>
+          >>>    # Get IAM integration object
+          >>>    iam_integration = [integration.uid for integration
+          >>>      in iam_integrations
+          >>>      if integration.name == "My S3 integration"][0]
+          >>>
+          >>>   # Set IAM integration for IAMIntegrtion object
+          >>>   dataset.set_iam_integration(iam_integration)
+          >>>
+          >>>   # Unset IAM integration
+          >>>   dataset.set_iam_integration()
+          """
+
+          # Unset IAM integration if iam_integration is None
+          if iam_integration is None:
+              query = """mutation DetachSignerPyApi($id: ID!) {
+                  clearSignerForDataset(where: {id: $id}) {
+                      id
+                      signer {
+                      id
+                      }
+                  }
+                  }"""
+              response = self.client.execute(query, {"id": self.uid})
+              
+              if response:
+                 return response["clearSignerForDataset"]["signer"]
+              else:
+                raise LabelboxError("Can't unset IAM integration")
+              
+          else:
+
+              if isinstance(iam_integration, IAMIntegration):
+                  iam_integration_id = iam_integration.uid
+              else:
+                  iam_integration_id = iam_integration
+
+              query = """mutation AttachSignerPyApi($signerId: ID!, $datasetId: ID!) {
+                  setSignerForDataset(data: {signerId: $signerId}, where: {id: $datasetId}) {
+                      id
+                      signer {
+                      id
+                      }
+                  }
+                  }"""
+              response = self.client.execute(query, {"signerId": iam_integration_id, "datasetId": self.uid})
+              
+              # Return IAM Integration object if  
```

Comment:
Delete this unfinished comment

Reply:
Or finish it; I just don't think it's super useful (presumably the complete comment is "Return object if it is created" or something equivalent)

Reply:
Comment removed